README.md

```diff
@@ -67,3 +67,23 @@ Forbid the use of multiline values to expansions.update.
 
 *Solution*: Don't do that. If you absolutely must, see the `files` param in the
 [documentation](https://github.com/evergreen-ci/evergreen/wiki/Project-Commands#expansions-update).
+
+### dependency-for-func
+Enforce that if a task contains a specified function, any dependencies required for
+that function are included as dependencies of that task.
+
+*Solution*: Add the required dependency to the task.
+
+#### Configuration
+
+Which dependencies should be enforced can be configured via the lint config files. The
+configuration should use the following format:
+
+```yaml
+rules:
+  - rule: dependency-for-func
+    dependencies:
+      func_name_0: [dependency_task_0, dependency_task_1]
```

Comment:
`dependency_task_0` -> `dependency_func_0`? Ditto elsewhere

Reply:
Per Slack discussion (because GH was down), the idea is that certain functions depend on tasks; we check for tasks that call these functions and ensure these tasks are correctly marking certain other tasks as dependencies.

This feels pretty intuitive to me after the discussion.

Reply:
Oh, if you'd like, consider adding a comment somewhere saying that this checks for function -> task dependency because dependencies are defined at the task level and apply only to tasks.